app_dart/index.yaml

```diff
@@ -8,6 +8,11 @@ indexes:
   properties:
   - name: CreateTimestamp
     direction: desc
+- kind: Checklist
```

Comment:
This will not create indexes for previous data. Is there a plan to create these indexes?

Explanation: https://stackoverflow.com/questions/35742786/how-to-add-a-composite-index-in-google-datastore

Reply:
In the past we've just had the person who updated the yaml file run the gcloud command to create the index.

Reply:
As talked about updating old data with 'master' below with Todd, this is of lower priority as long as those data is old enough, which actually is. I am gonna to merge this PR for now.

Created https://github.com/flutter/flutter/issues/52066 to track.